### ISSUE TITLE
Is the project still alive ?

### ISSUE THREAD

**Issue by julian70400** [flyspray/flyspray#815]
Hello,

No more website, no more communication, and still a lot of features missing. Is this project still alive ?

Thank you for your work.

Regards,

Ju

**Comment by HalibutGitWiz** [comment]
Would be interesting to know that as well. And it it's still alive, a broad idea of a the roadmap would be great too, since it's not possible to check that on the website anymore.

My company has been using Flyspray for more than 10 years and is really happy with it ! We miss some features though and might consider contributing, if the project is still going somewhere and if that somewhere is in the same direction as our ideas. :)

**Comment by Neustradamus** [comment]
@peterdd: It is possible to create a stable release?

The problem is that people do not want to update with "unstable" release...

I have already explained here:
- https://github.com/flyspray/flyspray/issues/805

**Comment by aducom** [comment]
The current codebase does not run under PHP 8.x, at least you get a ton of warnings. I was not able to fix all, and unfortunately i consider this (great) project dead...

**Comment by poucz** [comment]
if this is true, it would be appropriate to write it on the wiki page.

**Comment by peterdd** [comment]
I am still there, but currently have no resources.

https://groups.google.com/g/flyspray/c/RM4DIPBy-xc

**Comment by Thanatermesis** [comment]
@peterdd Thanks for the update, I still use flyspray too as the BTS for Elive: https://bugs.elivecd.org 

Unfortunately there's PHP8 errors, my hosting don't allow to continue using version 7.x, for example when registering a new user:

>  "Fatal error: Array and string offset access syntax with curly braces is no longer supported in /home/elivebugs/bugs.elivecd.org/vendor/dapphp/securimage/securimage.php on line 2209"

Specific line is:
```$letter = $code['display']{$i};```

I don't think is too difficult to grep & replace things like this one, but seems like the problem relies on that @peterdd and/or the devs cannot spend time on the project, I would like to know what would be the best way to keep this amazing project, useful to many, alive at least by other's code contributions 🤔 what do you suggest that will be best to proceed with, @peterdd ?

**Comment by peterdd** [comment]
Ok, I will look into this. I roughly tested PHP8 and seems to work, but I didn't tested this part.

**Comment by aducom** [comment]
There are many issues with PHP 8.1, I managed to fix a lot of them, but I continuously ran into new issues. One of them is:

Deprecated: Optional parameter $name declared before required parameter $wikiUri is implicitly treated as a required parameter in /home/aduc0madm/domains/phspeed.com/private_html/bt/plugins/dokuwiki/inc/parser/xhtml.php on line 575 Deprecated: Optional parameter $title declared before required parameter $wikiUri is implicitly treated as a required parameter in /home/aduc0madm/domains/phspeed.com/private_html/bt/plugins/dokuwiki/inc/parser/renderer.php on line 174 Deprecated: Optional parameter $mode declared before required parameter $matchPos is implicitly treated as a required parameter in /home/aduc0madm/domains/phspeed.com/private_html/bt/plugins/dokuwiki/inc/parser/lexer.php on line 441

Here I got lost in the code. But if it helps, I can provide the code with changes as I have it now.

**Comment by Thanatermesis** [comment]
Yes please submit your code contribution with these updates / fixes which can improve the default flyspray :) I think the easiest is from a PR way? 

@peterdd this error exactly seems to be happened when somebody tried to register into flyspray

**Comment by peterdd** [comment]
@Thanatermesis securimage upgraded to 3.6.8 in composer.json fixes this. 
https://github.com/flyspray/flyspray/commit/62e9c2e19bfdf6f3fb9e7aa9fe4b8269799d29ea

@aducom 

> Deprecated: Optional parameter $name declared before required parameter $wikiUri is implicitly treated as a required parameter in /home/aduc0madm/domains/phspeed.com/private_html/bt/plugins/dokuwiki/inc/parser/xhtml.php on line 575

This was fixed in master branch in Apr 2021. https://github.com/flyspray/flyspray/commit/7eda83837ed4c98ae173deedb7d5762273de11af

I hope I get bugs.flyspray.org up soon.

**Comment by aducom** [comment]
Are you really sure? My feeling is that I run the latest version, and it only appeared after updating to PHP 8.1. But to be sure I'll upload the latest branche.

**Comment by peterdd** [comment]
I made changes for php 8.x after 1.0-rc10.

A local test with current master branch with debian11 with php 8.2.7 and mariadb, dokuwiki as syntax and a backup of bugs.flyspray.org data:

![php8 2 7](https://github.com/flyspray/flyspray/assets/1839154/357f58eb-523f-432d-b59f-21507349c9e6)

**Comment by Neustradamus** [comment]
@peterdd: The problem is there is not a stable version, people do not install unstable build.

**Comment by aducom** [comment]
I'll see if I can install latest branch from git then.

**Comment by Thanatermesis** [comment]
confirmed that the registering works after an update of the composer modules

@peterdd what about releasing a new stable? i mean the actual dev versions works pretty good and much more featured than my previous "stable" versions :)

**Comment by aducom** [comment]
Unfortunately I run into issues: 

Unhandled exception: Call to undefined function get_magic_quotes_runtime()
This should never happen, please inform Flyspray Developers.

As I'm running in 8.x:

Warning

This function has been DEPRECATED as of PHP 7.4.0, and REMOVED as of PHP 8.0.0. Relying on this function is highly discouraged.

I installed the current branch so what am I doing wrong?


If you are an Administrator of this Flyspray installation you might enable temporarly! DEBUG_EXCEPTION in constants.inc.php for more details.

**Comment by aducom** [comment]
I did a scan and found two occurrences:
phpbrief.php and php.php in the flyspray-master\plugins\dokuwiki\inc\gehsi\

**Comment by peterdd** [comment]
> I did a scan and found two occurrences:
> phpbrief.php and php.php in the flyspray-master\plugins\dokuwiki\inc\gehsi\

Mhh, I don't think that's the reason as these 2 occurences are only quoted keywords for parsing text snippets for syntax highlighting content, not real php code.

How did you do the "scan"? With `grep -ri` or some other tool?

What exact Flyspray version had you before upgrade? mysql/mariadb or postgresql database? plain,dokuwiki or html/ckeditor syntax?

**Comment by aducom** [comment]
With total commander. My previous version was the last on your website including the vendor files (1.0 rc 10? can't tell the rc, as it is not in the upgrade folder (0.9.9; 0.9.9.2 /.4 /.5 ./7). I removed files according to the instructions and re-uploaded the vendor files as they are not in git.
I'm running on MariaDB in a shared hosting environment (https://www.phspeed.com/bt) Afaik ckeditor. 

b.r. Albert

**Comment by peterdd** [comment]
Have you tried **temporarly** uncommenting

`//define('DEBUG_EXCEPTION', true);`

`define('DEBUG_EXCEPTION', true);`

in includes/constants.inc.php to see more details of the error?

Usually the ./vendor/ files should also be up to date to the ./composer.json

Yeah, that I shipped with a release usually. As you are on a shared hosting you haven't ssh to run `composer update` or php composer.phar update?

**Comment by aducom** [comment]
No, that is why I usually use official releases. I cannot execute php -v and I did alter the debug, but I don't see more info. Probably that the provider has disabled. I'll aks. I'm not using composer so I probably don't have the latest vendor files.

**Comment by peterdd** [comment]
Ok, I understand. I had similiar conditions too. I need to do current status as release 1.0-rc11 and upload complete files here.

**Comment by aducom** [comment]
Sorry, I was not paying attention to the chat ;-)

**Comment by peterdd** [comment]
We could try a jitsi meeting?

**Comment by aducom** [comment]
Always prepared to assist, but that would have to be tomorrow or tonight, currently occupied with meetings.

**Comment by Neustradamus** [comment]
@Thanatermesis, @aducom, @poucz, @HalibutGitWiz, @julian70400: Flyspray 1.0 RC11 is out!
- https://github.com/flyspray/flyspray/releases/tag/v1.0-rc11

Time to update your version ^^
Please do a backup before.
Please inform if it is good for you or if you have a problem.
